curvefs/src/client/s3/disk_cache_manager.cpp

```diff
@@ -324,40 +335,32 @@ bool DiskCacheManager::IsDiskCacheSafe() {
 void DiskCacheManager::TrimCache() {
     const std::chrono::seconds sleepSec(trimCheckIntervalSec_);
     LOG(INFO) << "trim function start.";
+    waitIntervalSec_.Init(trimCheckIntervalSec_ * 1000);
     // 1. check cache disk usage every sleepSec seconds.
     // 2. if cache disk is full,
     //    then remove disk file until cache disk is lower than safeRatio_.
-    while (sleeper_.wait_for(sleepSec)) {
+    std::string cacheReadFullDir, cacheWriteFullDir,
+      cacheReadFile, cacheWriteFile, cacheKey;
+    cacheReadFullDir = GetCacheReadFullDir();
+    cacheWriteFullDir = GetCacheWriteFullDir();
+    while (true) {
+        waitIntervalSec_.WaitForNextExcution();
         if (!isRunning_) {
             LOG(INFO) << "trim thread end.";
             return;
         }
         VLOG(9) << "trim thread wake up.";
         InitQosParam();
         SetDiskFsUsedRatio();
-        if (IsDiskCacheFull()) {
-            VLOG(3) << "disk cache full, begin trim.";
-            std::string cacheReadFullDir, cacheWriteFullDir,
-              cacheReadFile, cacheWriteFile;
-            cacheReadFullDir = GetCacheReadFullDir();
-            cacheWriteFullDir = GetCacheWriteFullDir();
-            std::string cacheKey, cacheKeyBfo, cacheKeyDel;
-            if (!cachedObjName_->GetBack(&cacheKey)) {
-                VLOG(3) << "remove disk file error"
-                        << ", cachedObjName is empty.";
-                continue;
-            }
             while (!IsDiskCacheSafe()) {
-                cacheKeyDel = cacheKey;
-                if (!cachedObjName_->GetBefore(cacheKey, &cacheKeyBfo)) {
+                if (!cachedObjName_->GetFirst(false, &cacheKey)) {
```

Comment:
I don't think lru should do this

Reply:
have reached an agreement that it is not need to edit